pdesolvers/heat_equation_1d/solver/explicit_solver.py

```diff
@@ -0,0 +1,32 @@
+import numpy as np
+
+from pdesolvers.heat_equation_1d.strategy.solver_strategy import SolverStrategy
+
+
+class HeatEquationExplicitSolver(SolverStrategy):
+
+    def solve(self, context):
+
+        x = context._generate_x_grid()
+        dx = x[1] - x[0]
+
+        dt_max = 0.5 * (dx**2) / context._get_k()
```

Comment:
would it be better if i did it like this (e.g. automatically calculating the dt within the method to ensure stability)

or

do a check and tell the user that the inputted values are invalid (e.g. too large)?